runtime/autoload/provider/clipboard.vim

```diff
@@ -156,7 +156,14 @@ function! s:clipboard.get(reg) abort
   elseif s:selections[a:reg].owner > 0
     return s:selections[a:reg].data
   end
-  return s:try_cmd(s:paste[a:reg])
+
+  let clipboard_data = s:try_cmd(s:paste[a:reg])
+  if match(&clipboard, '\v(unnamed|unnamedplus)') >= 0 && get(s:selections[a:reg].data, 0, []) == clipboard_data
```

Comment:
bug: should use `==#`

Reply:
alternatively: could we make `==` mean `==` within script context? or would that be a bridge too far?

Reply:
or have a native Lua clipboard API? :P

Reply:
yes, the entire module could be translated to lua. PR:s welcome.

Reply:
Right case matching 🤦‍♂️ . Fixed in #14962 . At least I don't think anyone will trigger it unless intentionaly trying to :D

Is there any benefit in translating the entire file ? Maintainability ? I don't think it's performence critical part of grabing clipboard.